Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -478,13 +469,13 @@ def merge(args:)
 
       bottle = BottleSpecification.new
       bottle.root_url bottle_hash["bottle"]["root_url"]
-      cellar = bottle_hash["bottle"]["cellar"]
-      cellar = cellar.to_sym if any_cellars.include?(cellar)
-      bottle.cellar cellar
       bottle.prefix bottle_hash["bottle"]["prefix"]
       bottle.rebuild bottle_hash["bottle"]["rebuild"]
       bottle_hash["bottle"]["tags"].each do |tag, tag_hash|
-        bottle.sha256 tag_hash["sha256"] => tag.to_sym
+        cellar = tag_hash["cellar"]
+        cellar = cellar.to_sym if any_cellars.include?(cellar)
+        sha256_hash = { tag_hash["sha256"] => tag.to_sym, :cellar => cellar }
+        bottle.sha256 sha256_hash
```

Comment:
Would be great, if possible, to have some tests that touch these lines.

Reply:
These lines are already test by the integration tests we have in the bottle_spec.rb file, under the `describe "brew bottle --merge"`. The `...bottle.json` files contain cellars, and the merge_json_files above makes sure the cellar is not at the right level in the `bottle_hash` hash.

If we want to test this more we need to extract this to a separate function; but I think the current tests are enough here.

Reply:
> These lines are already test by the integration tests we have in the bottle_spec.rb file, under the `describe "brew bottle --merge"`.

Ah, it's because the integration test is Linux only. I think it's maybe worth making it on macOS too give this (I appreciate contradicting past me).

> If we want to test this more we need to extract this to a separate function; but I think the current tests are enough here.

Extraction would be fine with me 👍🏻